lib/ocamlorg_web/graphql.ml

```diff
@@ -0,0 +1,307 @@
+            ~args:Arg.[]
+            ~typ:(non_null (list (non_null deps)))
+            ~resolve:(fun _ p ->
+              let info = Package.info p in
+              get_deps info.Package.Info.dependencies)
+        ; field
+            "depopts"
+            ~doc:"The depopts of the package"
+            ~args:Arg.[]
+            ~typ:(non_null (list (non_null deps)))
+            ~resolve:(fun _ p ->
+              let info = Package.info p in
+              get_deps info.Package.Info.depopts)
+        ; field
+            "conflicts"
+            ~doc:"The conflicts of the package"
+            ~args:Arg.[]
+            ~typ:(non_null (list (non_null deps)))
+            ~resolve:(fun _ p ->
+              let info = Package.info p in
+              get_deps info.Package.Info.conflicts)
+        ; field
+            "url"
+            ~doc:"The url of the package"
+            ~args:Arg.[]
+            ~typ:url
+            ~resolve:(fun _ p ->
+              let info = Package.info p in
+              info.Package.Info.url)
+        ]))
+
+let packages_result =
+  Graphql_lwt.Schema.(
+    obj "allPackages" ~fields:(fun _ ->
+        [ field
+            "totalPackages"
+            ~doc:"total number of packages"
+            ~args:Arg.[]
+            ~typ:(non_null int)
+            ~resolve:(fun _ p -> p.total_packages)
+        ; field
+            "packages"
+            ~doc:"list of all packages"
+            ~args:Arg.[]
+            ~typ:(non_null (list (non_null package)))
+            ~resolve:(fun _ p -> p.packages)
+        ]))
+
+let schema : Dream.request Graphql_lwt.Schema.schema =
+  Graphql_lwt.Schema.(
+    schema
+      [ field
+          "allPackages"
+          ~typ:(non_null packages_result)
+          ~doc:
+            "Filter packages in ascending order by name or based on search \
+             query. Packages can also be paginated by setting the offset and \
+             limit as desired"
+          ~args:
+            Arg.
+              [ arg'
+                  ~doc:
+                    "Filter packages in ascending order by name by passing \
+                     sortByName or based on search query"
+                  "filter"
+                  ~typ:string
+                  ~default:"sortByName"
+              ; arg'
+                  ~doc:
+                    "Specifies at what index packages can start, set to 0 by \
+                     default which means start from the first package"
+                  "offset"
+                  ~typ:int
+                  ~default:0
+              ; arg'
+                  ~doc:
+                    "Specifies the limit which means the number of packages \
+                     you want to return if you do not want all packages \
+                     returned at once. By default, all the packages are \
+                     returned"
+                  "limit"
+                  ~typ:int
+                  ~default:0
```

Comment:
Can we make this an optional type i.e. 

```ocaml
arg
  ~doc:
    "Specifies the limit which means the number of packages \
     you want to return if you do not want all packages \
     returned at once. By default, all the packages are \
     returned"
  "limit"
  ~typ:int
```

And then below pattern-match on `limit` with:

```ocaml
match limit with 
  | None -> ... (* return all *)
  | Some limit -> ... (* limit packages *)
```

Using `0` here feels a bit wrong to me, if someone (for whatever reason...) made a query like `allPackages(offset: 20, limit: 0, filter: "sortByName")` I don't think they would expect all of the packages to be returned.